tests/Skills/Child/AdapterWithErrorHandler.cs

```diff
@@ -1,24 +1,93 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
 using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.BotBuilderSamples
 {
     public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
     {
-        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger)
+        private readonly BotState _conversationState;
+        private readonly BotState _userState;
+        private ConcurrentDictionary<string, string> _conversationMap = new ConcurrentDictionary<string, string>();
+
+        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger, ConversationState conversationState, UserState userState)
             : base(configuration, logger)
         {
-            OnTurnError = (turnContext, exception) =>
+            _conversationState = conversationState;
+            _userState = userState;
+
+            OnTurnError = async (turnContext, exception) =>
             {
                 // Log any leaked exception from the application.
-                logger.LogError($"Exception caught : {exception.Message}");
-                return Task.CompletedTask;
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encounted an error or bug.");
```

Comment:
We're in an error handler already. The odds are very good this SendActivity won't work, and will actually throw an exception preventing the code below from running.